### ISSUE TITLE
Pre-tokenized datasets don't work

### ISSUE THREAD

**Issue by mfirth-truffle** [axolotl-ai-cloud/axolotl#1945]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/axolotl-ai-cloud/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

When I provide a json and leave the type blank for datasets it should expect that the dataset is pre-tokenized and in {input_ids: [], attention_mask: [], labels: []} format

### Current behaviour

Errors out every time

### Steps to reproduce

Create an input ids, attention mask, labels file and use 
```
datasets:
  - path: processed_data.jsonl
    ds_type: json
    type: 
  ```

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.
